src/mame/drivers/h01x.cpp

```diff
@@ -325,6 +525,9 @@ READ8_MEMBER(h01x_state::mem_8000_r)
 				if(!BIT(offset, 5)) result &= ioport("LINE5")->read();
 				if(!BIT(offset, 6)) result &= ioport("LINE6")->read();
 				if(!BIT(offset, 7)) result &= ioport("LINE7")->read();
+				if(!BIT(offset, 8)) result &= ioport("LINE8")->read();
+				if(!BIT(offset, 9)) result &= ioport("LINE9")->read();
+				if(!BIT(offset, 10)) result &= ioport("LINE10")->read();
```

Comment:
Please change the code to use a `required_ioport_array` or an `optional_ioport_array` with `read_safe` so it doesn't have to do so many tag map lookups while running and gets validation benefits.